In [29]:
import numpy as np
import pandas as pd
from math import sqrt

1. Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy
Полученные значения должны быть равны.
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,
а затем с использованием функций из библиотек numpy и pandas.

In [3]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

$$cov_{XY} = M[(X - M(X))(Y - M(Y))] = M(XY) - M(X)M(Y)=\overline{X \cdot Y} - \overline{X} \cdot \overline{Y}$$

In [4]:
zp_mu = zp.mean()
zp_mu

101.4

In [5]:
ks_mu = ks.mean()
ks_mu

709.9

In [7]:
cov = np.mean(zp*ks) - np.mean(zp)*np.mean(ks)
cov

9157.839999999997

In [15]:
c = np.cov(zp, ks, ddof = 0)
c

array([[ 3494.64,  9157.84],
       [ 9157.84, 30468.89]])

Коэффициент корреляции Пирсона:

$$r_{XY} = \frac{cov_{XY}}{\sigma_{X}\sigma_{Y}}$$

Несмещенные:

In [20]:
s_zp = np.std(zp, ddof = 0)
s_zp

59.115480206118605

In [21]:
s_ks = np.std(ks, ddof = 0)
s_ks

174.55340157098058

In [36]:
r = c[0,1] / (s_zp * s_ks)
r

0.887490092073916

Смещенные:

In [23]:
c = np.cov(zp, ks, ddof = 1)
c

array([[ 3882.93333333, 10175.37777778],
       [10175.37777778, 33854.32222222]])

In [24]:
s_zp = np.std(zp, ddof = 1)
s_zp

62.31318747531162

In [25]:
s_ks = np.std(ks, ddof = 1)
s_ks

183.99544076477068

In [37]:
r = c[0,1] / (s_zp * s_ks)
r

0.887490092073916

2. Измерены значения IQ выборки студентов,
обучающихся в местных технических вузах:
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.
Известно, что в генеральной совокупности IQ распределен нормально.
Найдите доверительный интервал для математического ожидания с надежностью 0.95.

Несмещенная оценка дисперсии:

$$\sigma^2 = \frac{\sum\limits_{i=1}^{n} (x_i - \overline{x})^2}{n-1}$$

In [27]:
data = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])
s_nesm = np.std(data, ddof=1)
s_nesm

10.54566788359614

In [30]:
int_dn = sum(data)/data.size + (2.262 * s_nesm/sqrt(10))
int_dn

125.64339223691834

In [31]:
int_up = sum(data)/data.size - (2.262 * s_nesm/sqrt(10))
int_up

110.55660776308164

#### Доверительный интервал [110.557 ; 125.643 ]

3. Известно, что рост футболистов в сборной распределен нормально
с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,
среднее выборочное составляет 174.2. Найдите доверительный интервал для математического
ожидания с надежностью 0.95.

In [39]:
int_down = 174.2 - (1.96 * sqrt(25) / sqrt(27))
int_down

172.31398912064722

In [38]:
int_up = 174.2 + (1.96 * sqrt(25) / sqrt(27))
int_up

176.08601087935276

#### Доверительный интервал [172.314 ; 176.086]